glossary.yml

```
@@ -708,8 +708,10 @@
   en:
     term: "backward-compatible"
     def: >
-      Software which is able to be used the same way as earlier versions of itself
-      without problems.
+      A property of a system, hardware or software, that allows for interoperability
+      with an older legacy system, or with input designed for such a system. 
+      For example a function written in python 3 that can be run successfully with
+      python version 2 is backward-compatible.
```

```suggestion
      For example a [function](#function) written in [Python](#python) 3 that can be run successfully with
      Python version 2 is backward-compatible.
```

Add internal links and capitalise Python (because it is capitalised throughout the document). Note: Adding the link for function will mean this does not pass the linter until PR #221 is merged.